### ISSUE TITLE
new message notifications are not cleared / removed when a guild / party is deleted

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#7955]
When a private guild or party is deleted (i.e., when all members have left), any player who has an unread messages notification for that guild does not have the notification cleared properly. This is what remains in the player's account:

```
"newMessages": {
    "4ac79ea1-d9fe-475f-b9df-d24dcafb14f1": null,
},
```

What should happen is that the `newMessages` entry for the guild is deleted.

The result of this bug is that the count that appears on the notifications icon includes the null entry, so it's always too high by one. For example:

![image](https://cloud.githubusercontent.com/assets/1495809/18132319/36d213f0-6f4b-11e6-9e7a-2b96e1c41e6d.png)

Admins will have to do a migration remove the `null` notifications once this bug is fixed. The contributor who works on this doesn't have to worry about that, but if you could give us a reminder when you submit your PR, that would be helpful to make sure we don't forget. :)

**Comment by Alys** [comment]
Admins will have to do a migration remove the `null` notifications once this bug is fixed. I'm copying this comment to the top post.

**Comment by gambtho** [comment]
Question on this -- is it acceptable to have all of the guild/party messages removed any a user is removed from a group?  Also -- extra credit for writing the migration code too? :)

**Comment by khipkin** [comment]
@gambtho Can you rephrase your question please? I don't understand.  Are you asking if all messages from a user should be deleted if that user leaves a guild/party?

**Comment by gambtho** [comment]
not exactly -- i was asking if all unread messages associated with a guild/party to a given user should be deleted when that user leaves the group.

**Comment by khipkin** [comment]
@gambtho Ohh I see.  Interesting question!  I don't know the "correct" answer to that. @staff?

**Comment by khipkin** [comment]
Suggestion: [help welcome now] --> [in progress]
